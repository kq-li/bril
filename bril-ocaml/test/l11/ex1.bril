# ARGS: 10
@main(x: int) {
  zero: int = const 0;
  one: int = const 1;
  y: int = const 1;
.header:
  b: bool = gt x zero;
  br b .true .false;
.true:
  y: int = mul x y;
  x: int = sub x one;
  jmp .header;
.false:
  print y;
}
