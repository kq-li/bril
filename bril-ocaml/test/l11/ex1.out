1c1
< 3628800 total_dyn_inst: 50
---
> 3628800 total_dyn_inst: 56
